internal/backend/temporalclient/dataconverter.go

```diff
@@ -0,0 +1,90 @@
+package temporalclient
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"go.temporal.io/sdk/converter"
+)
+
+const (
+	configKeyEnvVarPrefix = "AK_DATACONV_ENCRYPTION_KEY_"
+)
+
+type DataConverterEncryptionConfig struct {
+	// If `Encrypt` is true, `KeyNames` must have at least one key name specified.
+	Encrypt bool `koanf:"encrypt"`
+
+	// Comma-separated list of key names.
+	// First key in the active encryption key. All the rest are for decryption only.
+	// For each name, the encryption key is fetched from the environment, named as
+	// "AK_DATACONV_ENCRYPTION_KEY_<name>". Each key is 64 characters long, hex-encoded.
```

Comment:
I don't see any checks that the user has indeed defined all the right keys (what *are* they right keys anyway?), and with valid values as described in this comment. This PR needs to check that and guide the user in a clear, user-friendly way in case of errors.

Reply:
code changed, now keys are in a single env var.